lib/engine/apps/exec_wrapper.js

```diff
@@ -129,7 +133,31 @@ class ExecWrapper extends ExecEnvironment {
         return deviceView.values();
     }
 
+    enterProcedure(procid, procname) {
+        // save the calling frame ID on the stack
+        this._procedureStack.push(this._procedureFrame);
+
+        // make a fresh ID for the new call
+        this._procedureFrameCounter++;
+        this._procedureFrame = this._procedureFrameCounter;
+    }
```

Comment:
This is the same implementation as the superclass right? We don't need to duplicate it.